src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -114,42 +155,92 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
 
             var templateContext = new TemplateContext
             {
-                OriginalTemplate = JObject.Parse(template),
+                OriginalTemplate = JObject.Parse(rootTemplate),
                 ExpandedTemplate = templatejObject,
                 IsMainTemplate = true,
                 ResourceMappings = armTemplateProcessor.ResourceMappings,
                 TemplateIdentifier = templateFilePath,
                 IsBicep = isBicep,
-                SourceMap = sourceMap
+                SourceMap = sourceMap,
+                PathPrefix = pathPrefix
             };
 
             try
             {
                 IEnumerable<IEvaluation> evaluations = this.jsonRuleEngine.AnalyzeTemplate(templateContext);
                 evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
 
-                // For each rule we don't want to report the same line more than once
-                // This is a temporal fix
-                var evalsToValidate = new List<IEvaluation>();
-                var evalsToNotValidate = new List<IEvaluation>();
-                foreach (var eval in evaluations)
+                // Recursively handle nested templates 
+                //dynamic jsonTemplate = JsonConvert.DeserializeObject(populatedTemplate);
+                var jsonTemplate = JObject.Parse(populatedTemplate);
+                dynamic processedTemplateResources = templatejObject["resources"];
+
+                for (int i = 0; i < processedTemplateResources.Count; i++)
                 {
-                    if (!eval.Passed && eval.Result != null)
-                    {
-                        evalsToValidate.Add(eval);
-                    }
-                    else
+                    var currentProcessedResource = processedTemplateResources[i];
+
+                    if (String.Equals(currentProcessedResource.type.Value, "Microsoft.Resources/deployments"))
```

Comment:
To do a case-insensitive comparison, that flag must be passed to this function.  The change in ArmTemplateProcessor.cs needs the update too.
```cs
                    if (String.Equals(currentProcessedResource.type.Value, "Microsoft.Resources/deployments", StringComparison.OrdinalIgnoreCase))
```